### PR TITLE
Cherry picks for release - 1.8.2

### PR DESCRIPTION
**Pull request by gwang-msft**
This PR contain the Cherry Picks for Release 1.8.2
rel-1.8.2 will be a patch release where only the iOS mobile package will be published.

Primarily the packaging changes for static framework for iOS
There is one additional change included
- Fix optimizer crash (#8274)\

### CODE REVIEW

**Review by edgchen1 — APPROVED** [review]

**Review by pranavsharma — APPROVED** [review]
Verified the optimizer fix.

**Comment by edgchen1** [comment]
Could we also include this one?
https://github.com/microsoft/onnxruntime/pull/8350

**Comment by snnn** [comment]
PR is ready.

File: tools/ci_build/github/azure-pipelines/orttraining-win-gpu-ci-pipeline.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ trigger: none
 jobs:
 - template: templates/win-ci-2019.yml
   parameters:
-    AgentPool : 'Win-GPU-2019'
+    AgentPool : 'onnxruntime-win-cuda11-0'
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/orttraining-win-gpu-ci-pipeline.yml:6]: The "Win-GPU-2019" machine pool doesn't have CUDA 11.0 anymore. It was removed when I added CUDA 11.1 in, to make room for the new one. 

So I created a new pool under name of "onnxruntime-win-cuda11-0" for this PR.

**Comment by gwang-msft** [comment]
Sending a new PR #8620 replacing this one, since there are additional cherry pick #8340 which has conflict with this PR

**Pull request closed without merging**